torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -267,6 +278,16 @@ c10::optional<Node*> tryMerge(
       getHeader(consumer),
       ":\n");
 
+  // This is a shortcut for Block code generation for single op.
+  // TODO: This needs to be cleaned up. Block code gen should use
+  // device type and not rely on a global flag.
+  if (tensorexpr::getTEGenerateBlockCode() &&
+      producer->kind() == prim::Constant &&
+      tensorexpr::isSupportedForBlock(consumer)) {
```

Comment:
This check probably belongs to `canHandle` function (see line 199 where we check if an op is supported by NNC). Are we doing this only because we cannot really merge ops yet and have to enable "fusion" for a single op? In a longer term it feels like the way fuser constructs subgraphs for fusion should be modified to support single-ops more easily (but if it's too much hassle for now I think it's fine with committing this PR first).

Reply:
Yes, that's right. We aren't merging ops for Block right now, and we needed a way to pass in a single op. When we starting supporting fusion, I'll modify this code.